src/Command/GenerateObjectAclCommand.php

```diff
@@ -48,11 +47,11 @@ final class GenerateObjectAclCommand extends QuestionableCommand
     private $aclObjectManipulators = [];
 
     /**
-     * @var ManagerRegistry|null
+     * @var ManagerRegistry
      */
     private $registry;
 
-    public function __construct(Pool $pool, array $aclObjectManipulators, ?ManagerRegistry $registry = null)
+    public function __construct(Pool $pool, array $aclObjectManipulators, ManagerRegistry $registry)
```

Comment:
This param should be nullable, otherwise you have to change command service definition:
```php
->set(GenerateObjectAclCommand::class, GenerateObjectAclCommand::class)
            ->public()
            ->tag('console.command')
            ->args([
                new ReferenceConfigurator('sonata.admin.pool'),
                [],
                (new ReferenceConfigurator('doctrine'))->nullOnInvalid(),
            ])
```

Reply:
I'm kinda confused because at the beginning of the execute function, there is:
```
if (!$this->registry) {
    throw new ServiceNotFoundException('doctrine', static::class, null, [], sprintf(
        'The command "%s" has a dependency on a non-existent service "doctrine".',
        static::$defaultName
    ));
}
```
So this command can't work without the `ManagerRegistry`.

Should I change 
```
(new ReferenceConfigurator('doctrine'))->nullOnInvalid(),
```
then ? Or is there a good reason to keep it nullable ?

Maybe @phansys can help, since he made the param nullable in the construct.

Reply:
I mean we should not register this command when `doctrine` is not found or we should keep it nullable.

Reply:
Imho we should not register the command if doctrine is not found. do you agree @sonata-project/contributors ?

Reply:
You probably dont want generate ACL list in production. It mean we should consider it like dev tool where throw expection about not existing service is acceptable. More, it will give advatage and devs will dont have to search why command is not existing. 

Imo both options are correct.

Reply:
Doctrine not found is more theoretical than a real use case right? Sonata cant be used without persistence bundle, and you will get doctrine installed when adding the persistence bundle. 

If that is the case I would simply go for not registering the command. If I am missing something else please explain

Reply:
Using DoctrineMongoDBBundle [there is no `doctrine` service](https://github.com/doctrine/DoctrineMongoDBBundle/blob/4.3.x/Resources/config/mongodb.xml#L182).

Taking a look at what the registry is for, looks like it is just used for creating the user class name:

https://github.com/sonata-project/SonataAdminBundle/blob/b5b2a66e826479234c81bb7d05523bd6d75dcdb4/src/Command/GenerateObjectAclCommand.php#L240-L247

because the `user_model` option expects `Shortcut notation like AcmeDemoBundle:User`, so maybe we could deprecate that in favor of asking for the FQCN `user_model` and there is no need for the registry. I could take a look maybe this evening.

Reply:
Ok I'll wait for the merge of your PR